src/runtime_src/core/edge/user/aie/aie.h

```diff
@@ -20,22 +20,54 @@
 #define xrt_core_edge_user_aie_h
 
 #include <vector>
+#include <queue>
 
+#include "core/edge/common/aie_parser.h"
 extern "C" {
 #include <xaiengine.h>
 }
 
 namespace zynqaie {
 
+#define BD_HIGH_ADDR_MASK               0xFF00000000L
+#define BD_LOW_ADDR_MASK                0xFFFFFFFF
+
+#define GET_BD_HIGH_ADDR(addr)          (((addr) & BD_HIGH_ADDR_MASK) >> 32)
+#define GET_BD_LOW_ADDR(addr)           ((addr) & BD_LOW_ADDR_MASK)
+
+struct BD {
+    uint16_t bd_num;
+    uint16_t addr_high;
+    uint32_t addr_low;
+};
+
+struct DMAChannel {
+    std::queue<BD> idle_bds;
+    std::queue<BD> pend_bds;
```

Comment:
queues are last resort.  I think I see that pend_bds must be a FIFO, so this is probably right.  But idle_bds, does the order matter?  If not make it a std::vector and use push(), back() and pop().